manticore/core/cpu/arm.py

```diff
@@ -900,6 +926,12 @@ def CMP(cpu, reg, compare):
     def POP(cpu, *regs):
         for reg in regs:
             val = cpu.stack_pop(cpu.address_bit_size / 8)
+            if reg.reg in ('PC', 'R15'):
+                if val & 0x1:
```

Comment:
How will this behave if `val` is symbolic? It might make sense to switch mode on executing, instead of switching prior to jumping. Might make forked states that end up in different exec modes easier to implement.